workers/unity/Packages/io.improbable.gdk.tools/GdkToolsConfigurationProvider.cs

```diff
@@ -35,40 +37,36 @@ public class GdkToolsConfigurationWindow : EditorWindow
         private List<string> configErrors = new List<string>();
         private Vector2 scrollPosition;
 
-        // Minimum time required from last config change before saving to file
-        private readonly TimeSpan FileSavingInterval = TimeSpan.FromSeconds(1);
-        private DateTime lastSaveTime = DateTime.Now;
+        // Flag to indicate if we have unsaved changes in the settings window
         private bool hasUnsavedData;
 
-        [MenuItem("SpatialOS/GDK tools configuration", false, MenuPriorities.GdkToolsConfiguration)]
-        public static void ShowWindow()
+        public GdkToolsConfigurationProvider(string path, SettingsScope scope = SettingsScope.User)
+        : base(path, scope) { }
+
+        [SettingsProvider]
+        public static SettingsProvider CreateGdkToolsConfigurationProvider()
         {
-            GetWindow<GdkToolsConfigurationWindow>().Show();
+            var provider = new GdkToolsConfigurationProvider("Project/GDK Tools Configuration", SettingsScope.Project);
+
+            PropertyInfo[] GdkToolsConfigProperties = typeof(GdkToolsConfiguration).GetProperties(BindingFlags.Public | BindingFlags.NonPublic | BindingFlags.Instance);
```

Comment:
This should use `var`

Reply:
A small comment on what this does would be useful. I assume this is generating keywords for the search bar?

Reply:
A note for this, the searching mostly works, except for `schema` which doesn't return any results despite there being a public field with schema in the name https://github.com/spatialos/gdk-for-unity/pull/1408/files#diff-99497508763024e2ab0873b449cca009R11

Not a huge deal, but interesting to note

Reply:
> I assume this is generating keywords for the search bar?

Yes, it should extract all the public and private class properties and use them as the keyword search terms.
Not sure why it isn't picking up the `SchemaSourceDirs` property from `GdkToolsConfiguration`...

Reply:
It turns out that the way I was doing it before wasn't picking up any of the properties (required for `CustomSnapshotPathPrefKey`), but I have updated the method so it now picks up all of the fields -- so all the search terms work now!